paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/clone/PickFilesForCloneOperator.java

```diff
@@ -95,6 +102,18 @@ public void processElement(StreamRecord<Tuple2<String, String>> streamRecord) th
         }
     }
 
+    private static Schema newSchemaFromTableSchema(TableSchema tableSchema) {
+        return new Schema(
+                ImmutableList.copyOf(tableSchema.fields()),
+                ImmutableList.copyOf(tableSchema.partitionKeys()),
+                ImmutableList.copyOf(tableSchema.primaryKeys()),
+                ImmutableMap.copyOf(
+                        Iterables.filter(
+                                tableSchema.options().entrySet(),
+                                entry -> !Objects.equals(entry.getKey(), CoreOptions.PATH.key()))),
+                tableSchema.comment());
```

Comment:
目前 clone 表创建新的 schema 时，会把原表中所有的 table options 都直接传入新的 schema 中，包括 source 表的 table path，然后在创建表时忽略它，这看起来不正确，并且在 FileSystemCatalog 中禁止传入自定义的 table path 后， clone table 的单侧会失败，所以我这里做了修复。

并且原来的 fromTableSchema 方法构建新 schema 时直接使用了 source schema 中的 partitionKeys 和 primaryKeys  实例，这并不安全，所以我这里做了拷贝。

同时我把这个方法从 Schema 类中移到了这里，这是唯一需要使用它的地方，如果你们觉得它足够通用，我可以把它放回去。

Currently, when creating a new schema for the clone table, all table options from the original table are directly passed into the new schema, including the table path of the source table. This is then ignored during table creation, which seems incorrect. Additionally, after prohibiting the passing of custom table paths in the FileSystemCatalog, the clone table's unit tests would fail. Therefore, I have made a fix here.

Moreover, the original `fromTableSchema` method, when constructing the new schema, directly used the `partitionKeys` and `primaryKeys` instances from the source schema, which is not safe. So, I have made a copy here.

I have also moved this method from the `Schema` class to here, as this is the only place where it is needed. If you feel it is sufficiently general, I can put it back.